scheduler/src/cook/mesos/mesos_compute_cluster.clj

```diff
@@ -14,14 +14,170 @@
 ;; limitations under the License.
 ;;
 (ns cook.mesos.mesos-compute-cluster
-  (:require [clojure.tools.logging :as log]
-            [cook.config :as config]
+  (:require [clojure.core.async :as async]
+            [clojure.data.json :as json]
+            [clojure.tools.logging :as log]
             [cook.compute-cluster :as cc]
+            [cook.config :as config]
+            [cook.mesos.heartbeat :as heartbeat]
+            [cook.mesos.sandbox :as sandbox]
             [cook.mesos.task :as task]
+            [cook.plugins.definitions :as plugins]
+            [cook.plugins.pool :as pool-plugin]
+            [cook.progress :as progress]
+            [cook.scheduler.scheduler :as sched]
             [datomic.api :as d]
-            [mesomatic.scheduler :as mesos]))
+            [mesomatic.scheduler :as mesos]
+            [metrics.meters :as meters]
+            [plumbing.core :as pc]))
+
+(meters/defmeter [cook-mesos scheduler mesos-error])
+(meters/defmeter [cook-mesos scheduler handle-framework-message-rate])
+(meters/defmeter [cook-mesos scheduler handle-status-update-rate])
+
+
+(defn create-mesos-scheduler
+  "Creates the mesos scheduler which processes status updates asynchronously but in order of receipt."
+  [gpu-enabled? conn heartbeat-ch pool->fenzo pool->offers-chan match-trigger-chan
+   handle-exit-code handle-progress-message sandbox-syncer-state compute-cluster]
+  (let [configured-framework-id (cook.config/framework-id-config)
```

Comment:
Something to either fix here or put in punch-list for later, but I believe that this cook.config/framework-id-config is one of the two remaining uses of this config variable. Might want to think about we're going to replace this.

One thought, we cast compute-cluster to MesosComputeCluster, then extract the :framework-id from it. THen this call goes away, and this code becomes (more) generic.